credentials/src/main/kotlin/web5/sdk/credentials/VerifiablePresentation.kt

```diff
@@ -0,0 +1,194 @@
+     *
+     * @param type The type of the presentation, as a [String].
+     * @param holder The holder URI of the presentation, as a [String].
+     * @param vcJwts The credentials used in the presentation, as a [String].
+     * @param additionalData The presentation data, as a generic mapping [Map<String, Any>].
+     * @return A [VerifiablePresentation] instance.
+     *
+     * Example:
+     * ```
+     *     val vp = VerifiablePresentation.create(
+     *       vcJwts = vcJwts,
+     *       holder = holderDid.uri,
+     *       type = "PresentationSubmission",
+     *       additionalData = mapOf("presentation_submission" to presentationSubmission)
+     *     )
+     * ```
+     */
+    @JvmOverloads
+    public fun create(
+      type: String? = null,
+      holder: String,
+      vcJwts: Iterable<String>,
+      additionalData: Map<String, Any>? = null
+    ): VerifiablePresentation {
+      val vpProperties: Map<String, Any> =
+        additionalData?.plus("verifiableCredential" to vcJwts) ?: mapOf("verifiableCredential" to vcJwts)
+
+      val vpDataModel = VpDataModel.builder()
+        .id(URI.create("urn:uuid:${UUID.randomUUID()}"))
+        .holder(URI.create(holder))
+        .properties(vpProperties)
+        .apply {
+          type?.let { type(it) }
+        }
+        .build()
+
+      return VerifiablePresentation(vpDataModel)
+    }
+
+    /**
+     * Verifies the integrity and authenticity of a Verifiable Presentation (VP) encoded as a JSON Web Token (JWT).
+     *
+     * This function performs several crucial validation steps to ensure the trustworthiness of the provided VP:
+     * - Verifies that the vcJwts inside the VP are valid
+     * - Parses and validates the structure of the JWT.
+     * - Ensures the presence of critical header elements `alg` and `kid` in the JWT header.
+     * - Resolves the Decentralized Identifier (DID) and retrieves the associated DID Document.
+     * - Validates the DID and establishes a set of valid verification method IDs.
+     * - Identifies the correct Verification Method from the DID Document based on the `kid` parameter.
+     * - Verifies the JWT's signature using the public key associated with the Verification Method.
+     *
+     * If any of these steps fail, the function will throw a [SignatureException] with a message indicating the nature of the failure.
+     *
+     * @param vpJwt The Verifiable Presentation in JWT format as a [String].
+     * @throws SignatureException if the verification fails at any step, providing a message with failure details.
+     * @throws IllegalArgumentException if critical JWT header elements are absent.
+     *
+     * ### Example:
+     * ```
+     * try {
+     *     VerifiablePresentation.verify(signedVpJwt)
+     *     println("VP Verification successful!")
+     * } catch (e: SignatureException) {
+     *     println("VP Verification failed: ${e.message}")
+     * }
+     * ```
+     */
+    public fun verify(vpJwt: String) {
```

Comment:
this might have some complexity. for example, I could make a presentation definition where I'm specifically requesting expired credentials. this check would fail, since JWT validation fails if the `exp` property is in the past

I don't think you need to worry about this now, but it may be worth adding a note

Reply:
hmm good to think about..
we could just simply throw in an `acceptExpired` flag but we should find all the edge cases